pkgs/development/tools/build-managers/fac/default.nix

```
@@ -0,0 +1,52 @@
+{ stdenv, pkgs, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  name = "fac-build";
+  version = "0.5.3";
+
+  src = fetchFromGitHub {
+    owner = "droundy";
+    repo = "fac";
+    rev = "05c8507e3e4c5484a6f0e1393cea0c37d1d679ae";
+    sha256 = "sha256:1gifrlb31jy8633rnhny58ccp3wlmd338129c6sh0h1a38vkmsxk";
+  };
+
+  patches = [
+    # fac includes a unit test called ls_files_works which assumes it's
+    # running in a git repo. Nix's sandbox runs cargo build outside git,
+    # so this test won't work.
+    ./skip-git-ls-files-test.patch
+  ];
+
+  # workaround for missing Cargo.lock file
+  cargoPatches = [ ./cargo-lock.patch ];
+
+  cargoSha256 = "sha256:1kk1d957h34z0z5xvzdrbjz99qddw20kpkrw7pn5nxnanyjavln5";
+
+  buildInputs = with pkgs; [ git ];
+
+  # fac calls git at runtime, expecting it to be in the PATH,
+  # so we need to patch it to call git by absolute path instead.
+  postPatch = ''
+    sed -i "/std::process::Command::new/ s:git:${pkgs.git}/bin/git:" src/git.rs
+  '';
+
+  meta = with stdenv.lib; {
+    description = ''
+      A build system that uses ptrace to handle dependencies automatically
+    '';
+    longDescription = ''
+      Fac is a general-purpose build system inspired by make that utilizes
+      ptrace to ensure that all dependences are enumerated and that all
+      source files are added to a (git) repo. An important feature of fac
+      is that it automatically handles dependencies, rather than either
+      complaining about them or giving an incorrect build. Currently, fac
+      only runs on linux systems, but on those systems it is incredibly
+      easy to use!
+    '';
+    homepage = "https://physics.oregonstate.edu/~roundyd/fac";
+    license = [ licenses.gpl2 ];
```

```suggestion
    license = licenses.gpl2Plus;
```

Should not be in a list. Also, `gpl2` is deprecated and should be `gpl2Only` or `gpl2Plus`. The readme states

> Fac is free software, and is licensed under the GNU General Public License, version 2 or later.

So, `gpl2Plus` is the correct license.